Parser/asdl_c.py

```diff
@@ -890,29 +904,29 @@ def visitModule(self, mod):
     return 0;
 }
 
-static int obj2ast_identifier(PyObject* obj, PyObject** out, PyArena* arena)
+static int obj2ast_identifier(PyObject* obj, PyObject** out, const char* field, PyArena* arena)
 {
     if (!PyUnicode_CheckExact(obj) && obj != Py_None) {
-        PyErr_SetString(PyExc_TypeError, "AST identifier must be of type str");
+        PyErr_Format(PyExc_TypeError, "field \\"%s\\" was expecting an AST identifier (must be of type str)", field);
```

Comment:
Maybe we could dro "AST identifier" and just say that we expected an object of type str or bytes. Many people won't understand where "AST identifier" comes from.

```
field \\"%s\\" was expecting an AST identifier (must be of type str)"
```

Reply:
Applied on the master